crates/cli-support/src/js/mod.rs

```diff
@@ -2873,33 +3048,63 @@ __wbg_set_wasm(wasm);"
                                 // Ignore the raw signature.
                                 None
                             }
                             AuxExportedMethodKind::Setter => {
+                                let class = match exported {
+                                    ClassOrNamespace::Class(ref mut class) => class,
+                                    ClassOrNamespace::Namespace(_) => {
+                                        bail!(
+                                            "the setter `{}` is not supported on `{}`. Enums only support static methods on them.",
+                                            name,
+                                            class
+                                        );
+                                    }
+                                };
+
                                 prefix += "set ";
                                 if export.generate_typescript {
                                     let location = FieldLocation {
                                         name: name.clone(),
-                                        is_static: receiver.is_static(),
+                                        is_static,
                                     };
                                     let accessor = FieldAccessor {
                                         ty: ts_arg_tys[0].clone(),
                                         docs: ts_docs.clone(),
                                         is_optional: might_be_optional_field,
                                     };
 
-                                    exported.push_accessor_ts(location, accessor, true);
+                                    class.push_accessor_ts(location, accessor, true);
                                 }
                                 None
                             }
                         };
 
-                        exported.push(name, &prefix, &js_docs, &code, &ts_docs, ts);
+                        match exported {
+                            ClassOrNamespace::Class(class) => {
+                                class.push(name, &prefix, &js_docs, &code, &ts_docs, ts);
+                            }
+                            ClassOrNamespace::Namespace(ns) => {
+                                if !is_static {
+                                    let msg = format!(
+                                        "The enum `{}` cannot support the instance method `{}`. \
+                                        No binding will be generated for this method. \
+                                        Consider moving the method in an `impl` block with the `#[wasm_bindgen]` attribute to avoid exporting it, \
+                                        or making it a static method by replacing `self` with `value: Self`.",
+                                        ns.name,
+                                        name
+                                    );
+                                    println!("WARNING: {}", msg);
```

Comment:
The warning has to happen in the proc-macro, not in the CLI.
Warnings in proc-macros are a bit tough, so its not a requirement for the PR.